apps/frontend/src/modules/chat/slices/chat.slice.ts

```diff
@@ -0,0 +1,147 @@
+import { createSlice, type PayloadAction } from "@reduxjs/toolkit";
+
+import { DataStatus } from "~/libs/enums/enums.js";
+import { type ValueOf } from "~/libs/types/types.js";
+import { type SelectedCategory } from "~/modules/categories/categories.js";
+import { ChatMessageAuthor, ChatMessageType } from "~/modules/chat/chat.js";
+import { type TaskCreateDto } from "~/modules/tasks/tasks.js";
+import { ButtonsModeOption } from "~/pages/chat/libs/enums/enums.js";
+
+import { checkIsTask } from "../libs/helpers/helpers.js";
+import {
+	type ChatMessageDto,
+	type TaskMessage,
+	type TextMessage,
+} from "../libs/types/types.js";
+import {
+	createTasksFromSuggestions,
+	getTasksForCategories,
+	initConversation,
+} from "./actions.js";
+
+type State = {
+	buttonsMode: ValueOf<typeof ButtonsModeOption>;
+	dataStatus: ValueOf<typeof DataStatus>;
+	messages: Omit<
+		ChatMessageDto<TaskMessage[] | TextMessage>,
+		"createdAt" | "id"
+	>[];
+	selectedCategories: SelectedCategory[];
+	taskSuggestions: TaskCreateDto[];
+	threadId: null | string;
+};
+
+const initialState: State = {
+	buttonsMode: ButtonsModeOption.NONE,
+	dataStatus: DataStatus.IDLE,
+	messages: [],
+	selectedCategories: [],
+	taskSuggestions: [],
+	threadId: null,
+};
+
+const { actions, name, reducer } = createSlice({
+	extraReducers(builder) {
+		builder
+			.addCase(createTasksFromSuggestions.pending, (state) => {
+				state.taskSuggestions = [];
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(createTasksFromSuggestions.fulfilled, (state) => {
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+			.addCase(createTasksFromSuggestions.rejected, (state) => {
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+
+			.addCase(initConversation.pending, (state) => {
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(initConversation.fulfilled, (state, action) => {
+				state.threadId = action.payload.threadId;
+
+				state.buttonsMode = ButtonsModeOption.SUGGESTIONS_CREATION;
+				state.dataStatus = DataStatus.FULFILLED;
+			})
+			.addCase(initConversation.rejected, (state) => {
+				state.dataStatus = DataStatus.REJECTED;
+			})
+
+			.addCase(getTasksForCategories.pending, (state) => {
+				state.dataStatus = DataStatus.PENDING;
+			})
+			.addCase(getTasksForCategories.fulfilled, (state, action) => {
+				state.dataStatus = DataStatus.FULFILLED;
+				const newMessages = action.payload.messages;
+				const taskSuggestionsMessagePayload: TaskMessage[] = [];
+
+				for (const message of newMessages) {
+					if (message.type === "text") {
+						state.messages.push({
+							author: ChatMessageAuthor.ASSISTANT,
+							isRead: true,
+							payload: message.payload as TextMessage,
+							type: message.type,
+						});
+					}
+
+					if (checkIsTask(message)) {
+						taskSuggestionsMessagePayload.push(message.payload);
+						state.taskSuggestions.push({
```

Comment:
Why do we need `taskSuggestions` in state? I believe it should be a part of messages

Reply:
These are needed to actually create the task/change task suggestions. The ones inside of messages are just for user to see